sympy/core/numbers.py

```diff
@@ -1059,11 +1059,9 @@ def __eq__(self, other):
         if isinstance(other, Float):
             return bool(mlib.mpf_eq(self._mpf_, other._mpf_))
         if isinstance(other, Number):
-            # numbers should compare at the same precision;
-            # all _as_mpf_val routines should be sure to abide
-            # by the request to change the prec if necessary; if
-            # they don't, the equality test will fail since it compares
-            # the mpf tuples
```

Comment:
We may want to add a new comment here, explaining the rationale (if we do decide to go with this change).